favor/favor.py

```diff
@@ -31,20 +31,24 @@ def __init__(
         self.f = f
         self.randomizer=randomizer
 
+        if orthonormal and m > key_dim:
+            raise ValueError('m <= key_dim is required if orthonormal == True')
+
         self._features = None
         self.register_buffer('phi_scale', torch.tensor(1./ math.sqrt(m)))
 
 
     def features(self):
         if self._features is None or self.redraw:
             self._features = self.randomizer(
-                (self.m, self.key_dim),
+                (self.key_dim, self.m),
                 device=self.phi_scale.device,
                 dtype=self.phi_scale.dtype
             )
             if self.orthonormal:
                 self._features = torch.qr(
                     self._features.double())[0].to(self.phi_scale.dtype)
+            self._features.t_()
```

Comment:
Why doing this and not (m, key_dim) in the first place?

Reply:
Because of this:
![image](https://user-images.githubusercontent.com/8046580/100384739-2ce21480-3021-11eb-82e9-e463f38e7a3b.png)

`torch.qr` will throw away columns if `m < key_dim`.

Reply:
I think it's because Gram-Schmidt normally works on columns, but we want rows.

Reply:
ok, great, good catch again then.
could you use `.transpose_()` instead of `.t_()` ? It looks like this one is not documented, or am I mistaken ?

Reply:
https://pytorch.org/docs/stable/tensors.html#torch.Tensor.t_

But I can use `.transpose_(0, 1)` if you prefer.

Reply:
ha, right. perfect !